File: abc/163/d/main.py
from collections import  deque
import sys
import bisect
import math
import itertools
from queue import deque
from math import gcd

INF = float('inf')
mod = 10**9+7
eps = 10**-7

def inp(): return int(sys.stdin.readline())
 
def inp_list(): return list(map(int, sys.stdin.readline().split()))

def lcm(x, y): return (x * y) // gcd(x, y)

N, K = inp_list()
ans = 0
for k in range(K, N + 2):
    # M => 和の最大値, m => 和の最小値, k について和としてありえるものの個数は M - m + 1
    M = ( N * (N + 1) // 2 - (N - k) * (N + 1 - k) // 2 )
    m = k * (k - 1) // 2
    ans += (M - m + 1)

print(ans % mod)

File: abc/163/d/main.go
package main

import (
	"fmt"
	"bufio"
	"os"
)

var (
	N, K int
	M, m int
)

func main () {
	r := bufio.NewReader(os.Stdin)
	fmt.Fscan(r, &N)
	fmt.Fscan(r, &K)

	mod := 1000000007
	ans := 0
	for k := K; k <= N + 1; k++ {
		M = sumOfNums(N - k + 1, N)
		m = sumOfNums(0, k - 1)
		ans += M - m + 1
		ans %= mod
	}
	fmt.Println(ans)
}

func sumOfNums(l, r int) int {
	return (l + r) * (r - l + 1) / 2
}
